src/EditorFeatures/Core/Implementation/ForegroundNotification/ForegroundNotificationService.cs

```diff
@@ -139,6 +147,15 @@ private void NotifyOnForegroundWorker()
                         {
                             // eat up cancellation
                         }
+                        catch (Exception ex) when (FatalError.ReportWithoutCrash(ex))
+                        {
+                            // The PendingWork callbacks should never throw.  In the case they do we
+                            // must ensure the IAsyncToken implementation is completed.  If it is not
```

Comment:
@jasonmalinowski if the IDE would prefer to crash here I'm very okay with that.  My change is an attempt to keep the existing semantics in place where throw is dealt with an an non-fatal case